### ISSUE TITLE
PhantomJS not found

### ISSUE THREAD

**Issue by leshinskii** [yt-dlp/yt-dlp#4851]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.09.01** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

yt-dlp not working through **proxy**
gives out:
PhantomJS not found

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
user@debian:~$ /media/user/XXxxxxxxxx/soft/yt-dlp  -vU --proxy socks5://user:pass@127.0.0.1:9150/ -R infinite --file-access-retries infinite --socket-timeout 3600 -N 20 https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9 https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9
[debug] Command-line config: ['-vU', '--proxy', 'socks5://user:pass@127.0.0.1:9150/', '-R', 'infinite', '--file-access-retries', 'infinite', '--socket-timeout', '3600', '-N', '20', 'https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9', 'https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d6] (zip)
[debug] Python 3.9.2 (CPython 64bit) - Linux-5.10.0-17-amd64-x86_64-with-glibc2.31 (glibc 2.31)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 4.3.4-0, ffprobe 4.3.4-0
[debug] Optional libraries: brotli-1.0.9, certifi-2020.06.20, sqlite3-2.6.0
[debug] Proxy map: {'http': 'socks5://user:pass@127.0.0.1:9150/', 'https': 'socks5://user:pass@127.0.0.1:9150/'}
[debug] Loaded 1670 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.09.01, Current version: 2022.09.01
yt-dlp is up to date (2022.09.01)
[debug] [PornHub] Extracting URL: https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9
[PornHub] ph626ad69c9a6b9: Downloading pc webpage
ERROR: [PornHub] ph626ad69c9a6b9: PhantomJS not found, Please download it from https://phantomjs.org/download.html
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/common.py", line 670, in extract
    ie_result = self._real_extract(url)
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/pornhub.py", line 279, in _real_extract
    webpage = dl_webpage('pc')
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/pornhub.py", line 275, in dl_webpage
    return self._download_webpage(
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/common.py", line 1079, in _download_webpage
    return self.__download_webpage(url_or_request, video_id, note, errnote, None, fatal, *args, **kwargs)
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/common.py", line 1030, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/pornhub.py", line 52, in _download_webpage_handle
    phantom = PhantomJSwrapper(self, required_version='2.0')
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/openload.py", line 115, in __init__
    raise ExtractorError(f'PhantomJS not found, {self.INSTALL_HINT}', expected=True)

[debug] [PornHub] Extracting URL: https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9
[PornHub] ph626ad69c9a6b9: Downloading pc webpage
ERROR: [PornHub] ph626ad69c9a6b9: PhantomJS not found, Please download it from https://phantomjs.org/download.html
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/common.py", line 670, in extract
    ie_result = self._real_extract(url)
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/pornhub.py", line 279, in _real_extract
    webpage = dl_webpage('pc')
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/pornhub.py", line 275, in dl_webpage
    return self._download_webpage(
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/common.py", line 1079, in _download_webpage
    return self.__download_webpage(url_or_request, video_id, note, errnote, None, fatal, *args, **kwargs)
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/common.py", line 1030, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/pornhub.py", line 52, in _download_webpage_handle
    phantom = PhantomJSwrapper(self, required_version='2.0')
  File "/media/user/XXxxxxxxxx/soft/yt-dlp/yt_dlp/extractor/openload.py", line 115, in __init__
    raise ExtractorError(f'PhantomJS not found, {self.INSTALL_HINT}', expected=True)
```

**Comment by leshinskii** [comment]
Everything works without using a proxy.
Don't install all node.js for me!
Please fix this!

**Comment by pukkandan** [comment]
Do what the error message says!

**Comment by leshinskii** [comment]
> Do what the error message says!

Managed to install PhantomJS
This is how it all works:
cd user/XXxxxxxxxx/soft/Downloads/3/phantomjs-2.1.1-linux-x86_64/bin && /media/user/XXxxxxxxxx/soft/yt-dlp  --proxy socks5://user:pass@127.0.0.1:9150/ -R infinite --file-access-retries infinite --socket-timeout 3600 -N 20 https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9

**Comment by leshinskii** [comment]
Alas, yesterday I made a mistake, and even when I put phantomjs in /usr/bin/ , it turned out that even if phantomjs is in place where it should be, the error still comes out, just different


/media/user/xxxxxxxxxxx/1/soft/yt-dlp -vU --proxy socks5://user:pass@127.0.0.1:9150/ -R infinite --file-access-retries infinite --socket-timeout 3600 -N 20 https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9
[debug] Command-line config: ['-vU', '--proxy', 'socks5://user:pass@127.0.0.1:9150/', '-R', 'infinite', '--file-access-retries', 'infinite', '--socket-timeout', '3600', '-N', '20', 'https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d6] (zip)
[debug] Python 3.9.2 (CPython 64bit) - Linux-5.10.0-17-amd64-x86_64-with-glibc2.31 (glibc 2.31)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 4.3.4-0, ffprobe 4.3.4-0, phantomjs 140669873113024
[debug] Optional libraries: brotli-1.0.9, certifi-2021.10.08, sqlite3-2.6.0
[debug] Proxy map: {'http': 'socks5://user:pass@127.0.0.1:9150/', 'https': 'socks5://user:pass@127.0.0.1:9150/'}
[debug] Loaded 1670 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.09.01, Current version: 2022.09.01
yt-dlp is up to date (2022.09.01)
[debug] [PornHub] Extracting URL: https://rt.pornhub.com/view_video.php?viewkey=ph626ad69c9a6b9
[PornHub] ph626ad69c9a6b9: Downloading pc webpage
ERROR: execute() takes from 2 to 3 positional arguments but 4 were given
Traceback (most recent call last):
  File "/media/user/xxxxxxxxxxx/1/soft/yt-dlp/yt_dlp/YoutubeDL.py", line 1459, in wrapper
    return func(self, *args, **kwargs)
  File "/media/user/xxxxxxxxxxx/1/soft/yt-dlp/yt_dlp/YoutubeDL.py", line 1535, in __extract_info
    ie_result = ie.extract(url)
  File "/media/user/xxxxxxxxxxx/1/soft/yt-dlp/yt_dlp/extractor/common.py", line 670, in extract
    ie_result = self._real_extract(url)
  File "/media/user/xxxxxxxxxxx/1/soft/yt-dlp/yt_dlp/extractor/pornhub.py", line 279, in _real_extract
    webpage = dl_webpage('pc')
  File "/media/user/xxxxxxxxxxx/1/soft/yt-dlp/yt_dlp/extractor/pornhub.py", line 275, in dl_webpage
    return self._download_webpage(
  File "/media/user/xxxxxxxxxxx/1/soft/yt-dlp/yt_dlp/extractor/common.py", line 1079, in _download_webpage
    return self.__download_webpage(url_or_request, video_id, note, errnote, None, fatal, *args, **kwargs)
  File "/media/user/xxxxxxxxxxx/1/soft/yt-dlp/yt_dlp/extractor/common.py", line 1030, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/media/user/xxxxxxxxxxx/1/soft/yt-dlp/yt_dlp/extractor/pornhub.py", line 53, in _download_webpage_handle
    phantom.get(url, html=webpage)
  File "/media/user/xxxxxxxxxxx/1/soft/yt-dlp/yt_dlp/extractor/openload.py", line 215, in get
    stdout = self.execute(jscode, video_id, note2)
TypeError: execute() takes from 2 to 3 positional arguments but 4 were given

**Comment by pukkandan** [comment]
Fixed by #4833

**Comment by leshinskii** [comment]
> Fixed by #4833

Hopefully a fixed version soon and I'll check if everything works well as it should

Will PhantomJS still be required to work through a proxy?
No proxy PhantomJS is required for yt-dlp to work
